apps/game_backend/lib/game_backend/campaigns.ex

```diff
@@ -24,7 +24,7 @@ defmodule GameBackend.Campaigns do
   def get_campaign(campaign_id) do
     case Repo.get(Campaign, campaign_id) |> Repo.preload(levels: [:units]) do
       nil -> {:error, :not_found}
-      campaign -> {:ok, campaign}
+      campaign -> {:ok, Map.put(campaign, :levels, Enum.sort_by(campaign.levels, & &1.level_number))}
```

Comment:
same as above, in this case it may be different because you are sorting a list inside the registers, it may be solved with a subquery but they are a bit hard to follow

Reply:
I don't think it'll be hard to see what the subquery does. The correct thing would be to handle it in the query just like the previous comment.